### PR TITLE
diagcfg : Read from global logging config if data dir not provided

### PR DESCRIPTION
**Pull request by bricerisingalgorand**
## Summary

When a datadir is not provided, this will have diagcfg read from the default logging config in the global config path. It will then generate a default config if that file does not exist

## Test Plan

I ran this locally and was able to produce correctly configured default logging configs

### CODE REVIEW

File: logging/telemetry.go [hunk 1]
```diff
@@ -92,11 +92,12 @@ func makeTelemetryState(cfg TelemetryConfig, hookFactory hookFactory) (*telemetr
...
 func ReadTelemetryConfigOrDefault(dataDir *string, genesisID string) (cfg TelemetryConfig, err error) {
 	err = nil
-	if dataDir != nil && *dataDir != "" {
+	dataDirProvided := dataDir != nil && *dataDir != ""
+	if dataDirProvided {
 		configPath := filepath.Join(*dataDir, TelemetryConfigFilename)
 		cfg, err = LoadTelemetryConfig(configPath)
 	}
-	if err != nil && os.IsNotExist(err) {
+	if err != nil && os.IsNotExist(err) || !dataDirProvided {
```
Comment (tsachiherman) [inline logging/telemetry.go:100]: The change is reasonable, however, I'd like to ask for few more small changes:
* change the function parameter from `*string` to `string`. Change these on the caller as well. ( there is no concrete current use case where we attempt to pass in nil )
* in the `if err != nil && os.IsNotExist(err) || !dataDirProvided` statement, make sure to use round brackets if there is more than a single logical expression. i.e. `if (a && b) || c
* Implement a unit test for this function. Unit test should fail prior to your change and pass with your change.

**Review by tsachiherman — CHANGES_REQUESTED** [review]
Change looks good, few small requests.

File: logging/telemetry_test.go [hunk 2]
```diff
@@ -18,6 +18,8 @@ package logging
 
 import (
 	"fmt"
+	"github.com/algorand/go-algorand/config"
```
Comment (tsachiherman) [inline logging/telemetry_test.go:21]: nit: coding style - the import list should be grouped as follows:
```
import (
 	"fmt"
 	"os"
 	"testing"
 	"time"

	"github.com/sirupsen/logrus"
	"github.com/stretchr/testify/require"

	"github.com/algorand/go-deadlock"

 	"github.com/algorand/go-algorand/config"
	"github.com/algorand/go-algorand/logging/telemetryspec"
)
```

**Review by tsachiherman — APPROVED** [review]

File: logging/telemetry_test.go (hunk 2 above)
Comment (bricerisingalgorand) [reply logging/telemetry_test.go]: Thanks! I just fixed that

**Review by tsachiherman — APPROVED** [review]
looks good.

**Pull request merged**